### PR TITLE
feat(useSorted): new function

### PR DESCRIPTION
**Pull request by okxiaoliang4**
### Description

Composable reactive sort array.

move from #1465

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useSorted/index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  compareFn: (a, b) => a.age - b.age,
+})
+const sorted = wrapFn(objArr)
+
+// or
+const wrapFn = useSortedWrapFn<User>(quickSort)
+const sorted = wrapFn(objArr, {
+  compareFn: (a, b) => a.age - b.age,
+})
+```
```
Comment (antfu) [inline packages/core/useSorted/index.md:49]: This has been removed right?

File: packages/core/useSorted/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  name: 'Joe',
+  age: 30,
+}, {
+  name: 'Jenny',
+  age: 22,
+}]
+const objSorted = useSorted(objArr, quickSort, {
+  compareFn: (a, b) => a.age - b.age,
+})
+```
```
Comment (antfu) [inline packages/core/useSorted/index.md:34]: I think it's better to explain what's `sortFn`, `compareFn` and `dirty` are for.

File: packages/core/useSorted/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+category: Sort
```
Comment (antfu) [inline packages/core/useSorted/index.md:2]:
```suggestion
category: Array
```

File: packages/core/useSorted/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}, {
+  name: 'Joe',
+  age: 30,
+}, {
+  name: 'Jenny',
+  age: 22,
+}]
+const objSorted = useSorted(objArr, quickSort, {
+  compareFn: (a, b) => a.age - b.age,
+})
```
Comment (antfu) [inline packages/core/useSorted/index.md:33]:
```suggestion
const objSorted = useSorted(objArr, (a, b) => a.age - b.age)
```

**Comment by antfu** [comment]
Converting to draft, please toggle when you back working on it.

**Comment by okxiaoliang4** [comment]
I am back! Too busy with work recently :(

File: packages/core/useSorted/index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+# useSorted
+
+reactive sort array
+
+## Usage
+
+```ts
+import { quickSort, useSorted } from '@vueuse/core'
```
Comment (antfu) [inline packages/core/useSorted/index.md:12]:
```suggestion
import { useSorted } from '@vueuse/core'
```

**Review by antfu — APPROVED** [review]

**Comment by CallerOfSouls** [comment]
I really think this is a neat composable, but I cannot for the life of me figure out how the compare function works to be able to do something like :
`[{name:"John",age:36,status:"First"},{name:"John",age:35,status:"First"},{name:"John",age:35,status:"waiting"},{name:"John",age:35,status:"attending"}]`

and then compare it so that I could sort it like `useSorted(items,(a,b)=>{
if(a.name == b.name)
   if(a.age == b.age)
        if(a.status > b.status){
        return position_in_array_to_be;
        }
    else if(a.name == b.name){
          if(a.age > b.age){
              return new_Position_value;
          }else{
if(a.status <= b.status){
    return position_in_array_to_be--;
    }
            return new_Position_Value++
          }
    }
})`. i am aware that this code does not work, I have tried several variations using a `var order = ref(0)`
and trying to keep up with an order by myself through addition and subtraction but there is no way to properly get that to work it seems. Any advice on how we could make this easier to use with multiple parameters would be awesome. Maybe I need to look into how to contribute myself..... but any help is appreciated again!

**Pull request merged**